### PR TITLE
Add angular 5 and replace AuthHttp with new Http Interceptor

### PR DESCRIPTION
**Pull request by rjwijnen**
Yesterday i tried to make the library work on Angular 5 and use the new HttpClient. As i saw a lot of other library's changed to the http interceptor for adding the auth. token. I tried to implement the logic from the old Auth Http into a new Interceptor. I hope some of you can double check everything is working?

### CODE REVIEW

**Comment by Lejdholt** [comment]
can't say if the implementation of the interceptor is correct or not, but i want it

**Comment by riisi** [comment]
Also looking to use this...

**Review by hkusulja — APPROVED** [review]
I confirm this for Angular 5

**Review by hkusulja — APPROVED** [review]

**Comment by hkusulja** [comment]
@sureshchahal almost two months passed, any info about merge for Angular 5 (current) support? Tx

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3145 @@
+{
```
Comment (ranveeraggarwal) [inline package-lock.json]: @rjwijnen you should probably not be pushing this file in.

**Comment by claudiuconstantin** [comment]
What's the status for this one?

**Comment by McTalian** [comment]
I have leveraged the AuthHttpInterceptor that is provided in this PR in my own project (copy and paste) and I can confirm that it works.

From what I see, there might just be a webpack warning due to dynamic requires if you were to try to use this with Angular 5, but this would at least enable folks to continue using this library with Angular 5.

Great work @rjwijnen , much appreciated!

**Comment by rjwijnen** [comment]
@ranveeraggarwal i removed the package lock.

**Comment by rjwijnen** [comment]
@sureshchahal could you look at this pull request and push it to master and NPM?

**Comment by jmtrz** [comment]
Hi Is this already fix i am getting this error 

WARNING in ./node_modules/ng2-adal/node_modules/@angular/core/@angular/core.es5.js
5659:15-36 Critical dependency: the request of a dependency is an expression

WARNING in ./node_modules/ng2-adal/node_modules/@angular/core/@angular/core.es5.js
5675:15-102 Critical dependency: the request of a dependency is an expression

If already fxed may i know how will i implement it in my proj 👍  tnx

**Comment by saransh944** [comment]
@rjwijnen can you please create your own project for the version 5 changes and host it on NPM? Looks like @sureshchahal is not active on this anymore.

**Comment by sureshchahal** [comment]
Yes please, sorry I have lot at hand at moment. If someone wants to take
ownership , I am happy to add them as owner also.

**Comment by saransh944** [comment]
Thanks for the response @sureshchahal . Just a quick note, a lot of users of this repo including myself reviewed the changes by @rjwijnen and everything looks good. Is it possible for you to merge this change?

**Comment by rjwijnen** [comment]
@saransh944  in the meantime i switched to https://github.com/benbaran/adal-angular4 ..

**Comment by saransh944** [comment]
@rjwijnen , thanks for the suggestion. Since we are talking about this, how are you handling the behavior that happens after the token is refreshed? I have noticed whenever the token refreshes, app is initialized 2-3 times, also Angular change detection does not work properly, i.e. ngModule breaks, @input changes from parent to child component doesn't propagate, etc. I am deleting the adal iFrame that gets created for the token refresh in case that is the reason for this weird behavior, but still I'm seeing these glitches. Do you have any recommendations on that?

**Pull request merged**